Kernel/Heap/kmalloc.cpp

```diff
@@ -243,6 +298,33 @@ void* kmalloc_eternal(size_t size)
     return ptr;
 }
 
+void* kmalloc_eternal(size_t size)
+{
+    kmalloc_verify_nospinlock_held();
+
+    size = round_up_to_power_of_two(size, sizeof(void*));
+
+    ScopedSpinLock lock(s_lock);
+    // Note: Only Eternal heaps should be in the static eternal heap of the kernel,
+    // therefore each pointer must of an EternalHeap and only it!
+    EternalHeap* eternal_heaps_metadata = reinterpret_cast<EternalHeap*>(kmalloc_eternal_heaps);
+    if (auto* ptr = eternal_heaps_metadata[g_dynamic_eternal_heaps_count].allocate(size); ptr != nullptr)
+        return ptr;
+
+    // Note: It's very unlikely that the MemoryManager was not initialized at this stage!
+    VERIFY(MemoryManager::is_initialized());
+
+    // Allocate a region. If the allocation fails, assert.
+    auto region = MM.allocate_kernel_region((3 * MiB), "kmalloc eternal heap", Region::Access::Read | Region::Access::Write, AllocationStrategy::AllocateNow);
```

Comment:
Let's not allocate 3 MiB at a time, that seems overly wasteful!

How about 1 MiB?

Reply:
Seems reasonable to allocate only 1 MiB.